### PR TITLE
python39Packages.google-auth-oauthlib: 0.4.6 -> 0.5.1

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 166062` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.dm-sonnet</li>
<li>python39Packages.graph_nets</li>
<li>python39Packages.optuna</li>
<li>python39Packages.pywick</li>
<li>python39Packages.rl-coach</li>
<li>python39Packages.shap</li>
<li>python39Packages.skorch</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python39Packages.elegy: log was empty</li>
<li>python39Packages.treex: log was empty</li>
  </ul>
</details>
<details>
<summary>
41 packages built:
</summary>
<ul>
<li>beancount (python39Packages.beancount)</li>
<li>fava</li>
<li>git-annex-remote-googledrive</li>
<li>home-assistant-component-tests.nest</li>
<li>ledger2beancount</li>
<li>python310Packages.beancount</li>
<li>python310Packages.beancount_docverif</li>
<li>python310Packages.drivelib</li>
<li>python310Packages.gcsfs</li>
<li>python310Packages.google-auth-oauthlib</li>
<li>python310Packages.google-nest-sdm</li>
<li>python310Packages.gspread</li>
<li>python310Packages.oauthenticator</li>
<li>python39Packages.augmax</li>
<li>python39Packages.beancount_docverif</li>
<li>python39Packages.boxx</li>
<li>python39Packages.bpycv</li>
<li>python39Packages.dm-haiku</li>
<li>python39Packages.drivelib</li>
<li>python39Packages.einops</li>
<li>python39Packages.ffcv</li>
<li>python39Packages.flax</li>
<li>python39Packages.gcsfs</li>
<li>python39Packages.google-auth-oauthlib</li>
<li>python39Packages.google-nest-sdm</li>
<li>python39Packages.gspread</li>
<li>python39Packages.ignite</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.objax</li>
<li>python39Packages.optax</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-machine-learning</li>
<li>python39Packages.rising</li>
<li>python39Packages.slicer</li>
<li>python39Packages.spacy-transformers</li>
<li>python39Packages.stanza</li>
<li>python39Packages.test-tube</li>
<li>python39Packages.torch-tb-profiler</li>
<li>python39Packages.torchinfo</li>
<li>python39Packages.zcs</li>
<li>s3ql</li>
</ul>
</details>

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 166062` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.pywick</li>
    <li>python39Packages.rl-coach</li>
    <li>python39Packages.shap</li>
    <li>python39Packages.skorch</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python39Packages.elegy</li>
    <li>python39Packages.pytorch-lightning</li>
    <li>python39Packages.pytorch-metric-learning</li>
    <li>python39Packages.pytorchWithCuda</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.treex</li>
  </ul>
</details>
<details>
  <summary>61 packages built:</summary>
  <ul>
    <li>beancount (python39Packages.beancount)</li>
    <li>fava</li>
    <li>git-annex-remote-googledrive</li>
    <li>gpt2tc</li>
    <li>home-assistant-component-tests.nest</li>
    <li>ledger2beancount</li>
    <li>libtensorflow</li>
    <li>python310Packages.beancount</li>
    <li>python310Packages.beancount_docverif</li>
    <li>python310Packages.drivelib</li>
    <li>python310Packages.gcsfs</li>
    <li>python310Packages.google-auth-oauthlib</li>
    <li>python310Packages.google-nest-sdm</li>
    <li>python310Packages.gspread</li>
    <li>python310Packages.oauthenticator</li>
    <li>python39Packages.augmax</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.beancount_docverif</li>
    <li>python39Packages.boxx</li>
    <li>python39Packages.bpycv</li>
    <li>python39Packages.dm-haiku</li>
    <li>python39Packages.drivelib</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.ffcv</li>
    <li>python39Packages.flax</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.google-nest-sdm</li>
    <li>python39Packages.gspread</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.mask-rcnn</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.objax</li>
    <li>python39Packages.optax</li>
    <li>python39Packages.pyro-ppl</li>
    <li>python39Packages.pytorch (python39Packages.pytorchWithoutCuda)</li>
    <li>python39Packages.pytorch-pfn-extras</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-machine-learning</li>
    <li>python39Packages.rising</li>
    <li>python39Packages.scikit-tda</li>
    <li>python39Packages.slicer</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.stanza</li>
    <li>python39Packages.tensorboardx</li>
    <li>python39Packages.tensorflow (python39Packages.tensorflow-build ,python39Packages.tensorflowWithoutCuda)</li>
    <li>python39Packages.tensorflow-bin</li>
    <li>python39Packages.tensorflow-datasets</li>
    <li>python39Packages.tensorflow-probability</li>
    <li>python39Packages.tensorflow-tensorboard</li>
    <li>python39Packages.test-tube</li>
    <li>python39Packages.tflearn</li>
    <li>python39Packages.torch-tb-profiler</li>
    <li>python39Packages.torchgpipe</li>
    <li>python39Packages.torchinfo</li>
    <li>python39Packages.torchvision</li>
    <li>python39Packages.umap-learn</li>
    <li>python39Packages.zcs</li>
    <li>s3ql</li>
    <li>tts</li>
  </ul>
</details>

**Comment by samuela** [comment]
Here are the failure logs:
```
error: builder for '/nix/store/br979lsywblaxl1wk7ifx642klj1jrls-python3.9-pytorch-1.10.2.drv' failed with exit code 1;
       last 10 log lines:
       > CMake Error at cmake/public/cuda.cmake:42 (message):
       >   PyTorch requires CUDA 10.2 or above.
       > Call Stack (most recent call first):
       >   cmake/Dependencies.cmake:1191 (include)
       >   CMakeLists.txt:653 (include)
       >
       >
       > -- Configuring incomplete, errors occurred!
       > See also "/build/source/build/CMakeFiles/CMakeOutput.log".
       > See also "/build/source/build/CMakeFiles/CMakeError.log".
       For full logs, run 'nix log /nix/store/br979lsywblaxl1wk7ifx642klj1jrls-python3.9-pytorch-1.10.2.drv'.
[1/11/63 built (1 failed)] building python3.9-pytorch-1.10.2 (buildPhase):       |         ^~~~~~~
[1/11/63 built (1 failed)] building python3.9-pytorch-1.10.2 (buildPhase):       |       ^~~~
error: builder for '/nix/store/ghwfxkh7flkk4cgg6bjq6rs87ssrx2kl-python3.9-pytorch-lightning-1.5.8.drv' failed with exit code 1;
       last 10 log lines:
       > removing build/bdist.linux-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./pytorch_lightning-1.5.8-py3-none-any.whl
       > Requirement already satisfied: torch>=1.7.* in /nix/store/s3sj6sz9fcrmrd8ggpb058f6lksbqwla-python3.9-pytorch-1.10.2/lib/python3.9/site-packages (from pytorch-lightning==1.5.8) (1.10.2)
       > Requirement already satisfied: tqdm>=4.41.0 in /nix/store/vl86xapfvh98nid5vs72xr055alwal3k-python3.9-tqdm-4.62.3/lib/python3.9/site-packages (from pytorch-lightning==1.5.8) (4.62.3)
       > ERROR: Could not find a version that satisfies the requirement torchmetrics>=0.4.1 (from pytorch-lightning) (from versions: none)
       > ERROR: No matching distribution found for torchmetrics>=0.4.1
       For full logs, run 'nix log /nix/store/ghwfxkh7flkk4cgg6bjq6rs87ssrx2kl-python3.9-pytorch-lightning-1.5.8.drv'.
error: builder for '/nix/store/mj59jp226kalrra5pl886qqg3bzmf6ig-python3.9-treeo-0.4.0.drv' failed with exit code 1;
       last 10 log lines:
       > Successfully built treeo
       > Finished creating a wheel...
       > Finished executing pipBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./treeo-0.0.4-py3-none-any.whl
       > Requirement already satisfied: typing-extensions in /nix/store/7dlrj4c3cv1r9n7zhmillxikm25gd84j-python3.9-typing-extensions-4.0.1/lib/python3.9/site-packages (from treeo==0.0.4) (4.0.1)
       > ERROR: Could not find a version that satisfies the requirement jax<0.3.0,>=0.2.18 (from treeo) (from versions: none)
       > ERROR: No matching distribution found for jax<0.3.0,>=0.2.18
       For full logs, run 'nix log /nix/store/mj59jp226kalrra5pl886qqg3bzmf6ig-python3.9-treeo-0.4.0.drv'.
error: builder for '/nix/store/p0pv3k1x54cnvxhlns17b89n14947gwk-python3.9-pytorch-metric-learning-1.1.0.drv' failed with exit code 2;
       last 10 log lines:
       >
       > src/pytorch_metric_learning/utils/common_functions.py:396
       >   /build/source/src/pytorch_metric_learning/utils/common_functions.py:396: DeprecationWarning: invalid escape sequence \.
       >     epoch_regex = regex_wrapper("[0-9]+\.pth$")
       >
       > -- Docs: https://docs.pytest.org/en/stable/warnings.html
       > =========================== short test summary info ============================
       > ERROR tests/utils/test_inference.py
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > ================== 7 deselected, 3 warnings, 1 error in 1.62s ==================
       For full logs, run 'nix log /nix/store/p0pv3k1x54cnvxhlns17b89n14947gwk-python3.9-pytorch-metric-learning-1.1.0.drv'.
error: builder for '/nix/store/5jfaiiiz9vx8w6mpwyzsd4x04s390ldx-tensorflow-gpu-2.7.1.drv' failed with exit code 1;
       last 10 log lines:
       > [14,235 / 20,040] Compiling external/nccl_archive/src/collectives/device/max_i8_all_reduce.cu.cc; 71s local ... (128 actions, 127 running)
       > [15,109 / 20,040] Compiling external/nccl_archive/src/collectives/device/sum_u8_all_reduce.cu.cc; 59s local ... (128 actions, 127 running)
       > [15,856 / 20,040] Compiling tensorflow/compiler/mlir/tensorflow/ir/tf_ops_a_m.cc; 102s local ... (128 actions, 127 running)
       > [16,073 / 20,040] Compiling tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc; 152s local ... (128 actions, 127 running)
       > ERROR: /build/source/tensorflow/core/data/service/BUILD:266:11: C++ compilation of rule '//tensorflow/core/data/service:dispatcher_client' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/k8-opt/bin/tensorflow/core/data/service/_objs/dispatcher_client/dispatcher_client.pic.d ... (remaining 228 argument(s) skipped)
       > INFO: Elapsed time: 500.288s, Critical Path: 187.26s
       > INFO: 16362 processes: 6877 internal, 9485 local.
       > FAILED: Build did NOT complete successfully
       > FAILED: Build did NOT complete successfully
       >
       For full logs, run 'nix log /nix/store/5jfaiiiz9vx8w6mpwyzsd4x04s390ldx-tensorflow-gpu-2.7.1.drv'.
```
Doesn't look like any of them are related to this PR.

**Review by samuela — APPROVED** [review]
LGTM Thanks for following up @SuperSandro2000!

Sorry for the confusion regarding the breakage and reversion. I try to work hard to keep the deep learning packages running smoothly in nixpkgs, so I responded more quickly than usual. I appreciate your feedback regarding the process though, and next time I'll wait a day or two before reverting to give you more time to respond!
